packages/api/test/repository-test.js

```diff
@@ -59,6 +59,47 @@ describe('Repository', () => {
     });
   });
 
+  describe('ticket handlers', async () => {
+    const state = 'testing-state';
+    const ticket = {
+      state: state,
+      data: 'fake data'
+    };
+    describe('storeTicket', async () => {
+      it('stores a ticket which has a state', async () => {
+      // Store ticket
+        await repo.storeTicket(ticket);
+
+        // Retrieve ticket, check it
+        const result = await repo.getDoc('tickets/' + state);
+        assert.deepStrictEqual(result.ticket, ticket);
+
+        // Delete ticket
+        await repo.deleteDoc('tickets/' + state);
+      });
+    });
+
+    describe('getTicket', async () => {
+      it('retrieves a ticket by its state', async () => {
+        await repo.createDoc('tickets/' + state, {
+          ticket: ticket
+        });
+
+        const result = await repo.getTicket(state);
+
+        assert.deepStrictEqual(result, ticket);
+
+        await repo.deleteDoc('tickets/' + state);
+      });
+    });
+  });
+
+  describe('actionIsAllowed', () => {
+    it('permits test deletion when the userr is not an admin but has access to the repo', () => {
+      const res = repo.actionIsAllowed([{ full_name: 'fake-org/fake-repo' }], 'fake-org/fake-repo', 'delete-test');
```

Comment:
I think you could potentially use a similar approach here with the mocking we did last week, where you mock the step that returns the permission object from GitHub. I would test with the real thing first, and make sure we understand what responses are possible.